arangosh/Shell/V8ClientConnection.cpp

```diff
@@ -297,6 +337,8 @@ void V8ClientConnection::connect() {
 void V8ClientConnection::reconnect() {
   std::lock_guard<std::recursive_mutex> guard(_lock);
 
+  std::string id = connectionIdentifier(_builder);
```

Comment:
This is the id of the old connection (with the old `_builder` settings), so this should be reflected in the name.  Perhaps we can also move the code to swap out the old connection up here, so that these logically connected parts are kept together.